### PR TITLE
base.yaml add e2fsprogs

### PR DESCRIPTION
**Pull request by amfern**
Please add the following dependency to the rosdep database.

## Package name:
libext2fs-dev

## Package Upstream Source:
https://sourceforge.net/projects/e2fsprogs/

## Purpose of using this:
This dependency is used to interact with drive and it would be really nice to install it with rosdep

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bullseye/libext2fs-dev
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/focal/libext2fs-dev
- Fedora: https://src.fedoraproject.org/browse/projects/
  - https://src.fedoraproject.org/rpms/e2fsprogs
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/core/x86_64/e2fsprogs/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/sys-libs/e2fsprogs-libs
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/e2fsprogs#default
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://pkgs.alpinelinux.org/package/edge/main/x86_64/e2fsprogs
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=21.05&show=e2fsprogs&from=0&size=50&sort=relevance&type=packages&query=e2fsprogs

Galactic

# The source is here: 
https://sourceforge.net/projects/e2fsprogs/

# Checks
 - [ ] All packages have a declared license in the package.xml
 - [ ] This repository has a LICENSE file
 - [ ] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Comment by wjwwood** [comment]
Please address the CI failure, it looks like you have some things out of alphabetical ordering:

https://github.com/ros/rosdistro/pull/30769/checks?check_run_id=3648304979#step:6:183

**Comment by amfern** [comment]
sorted packages in alphabetical order and added the CI suggested packages

**Review by mabelzhang — COMMENTED** [review]
Thank you for providing the links! That helps to speed up the review.
Are `e2fsprogs` and `libext2fs` equivalent? Which ones did CI suggest you to add - there's no diff for that second commit since it was a force push. Sometimes CI suggests things that need to be double-checked by human.

My guess is that because `libext2fs` exists only on Debian, OpenSUSE, and Ubuntu, the `e2fsprogs` for all the other distros are equivalent alternatives? But Debian, OpenSUSE, and Ubuntu also have `esprogs` packages (below). That's why I'm double-checking whether the two are equivalent.
Debian: https://packages.debian.org/bullseye/e2fsprogs
OpenSUSE: https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/e2fsprogs-devel-1.46.4-1.1.x86_64.rpm.html
Ubuntu: https://packages.ubuntu.com/focal/e2fsprogs

**Comment by amfern** [comment]
That is a good point, it seems like debian, ubuntu and OpenSUSE split the `e2fsprogs` package and `libext2fs2` is one of them. Other distors just install the `e2fsprogs` as a whole. I guess I should change this PR to adding `e2fsprogs` instead of `libext2fs2`

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2865,6 +2865,17 @@ libedit-dev:
   nixos: [libedit]
   rhel: [libedit-devel]
   ubuntu: [libedit-dev]
+e2fsprogs:
+  alpine: [e2fsprogs]
+  arch: [e2fsprogs]
+  debian: [e2fsprogs]
+  fedora: [e2fsprogs]
+  gentoo: [sys-libs/e2fsprogs-libs]
```
Comment (cottsay) [inline rosdep/base.yaml:2873]: This looks like the libraries, but I think you want the utilities: https://packages.gentoo.org/packages/sys-fs/e2fsprogs
```suggestion
  gentoo: [sys-fs/e2fsprogs]
```

File: rosdep/base.yaml [hunk 2]
```diff
@@ -2865,6 +2865,17 @@ libedit-dev:
...
+  macports: [e2fsprogs]
+  nixos: [e2fsprogs]
+  opensuse: [e2fsprogs-libs]
+  rhel: [e2fsprogs-devel]
```
Comment (cottsay) [inline rosdep/base.yaml:2877]: This provides the headers, but I think you want the utilities: http://mirror.centos.org/centos/8/BaseOS/x86_64/os/Packages/e2fsprogs-1.45.6-1.el8.x86_64.rpm
```suggestion
  rhel: [e2fsprogs]
```

File: rosdep/base.yaml [hunk 3]
```diff
@@ -2865,6 +2865,17 @@ libedit-dev:
...
   ubuntu: [libedit-dev]
+e2fsprogs:
+  alpine: [e2fsprogs]
+  arch: [e2fsprogs]
+  debian: [e2fsprogs]
+  fedora: [e2fsprogs]
+  gentoo: [sys-libs/e2fsprogs-libs]
+  macports: [e2fsprogs]
+  nixos: [e2fsprogs]
+  opensuse: [e2fsprogs-libs]
```
Comment (cottsay) [inline rosdep/base.yaml:2876]: This looks like the libraries, but I think you want the utilities: https://software.opensuse.org/package/e2fsprogs

```suggestion
  opensuse: [e2fsprogs]
```

**Review by cottsay — DISMISSED** [review]
If I understand your intentions correctly, you want the command line utilities, right? It looks like that's what the ubuntu package you're adding contains: https://packages.ubuntu.com/focal/amd64/e2fsprogs/filelist

I think that more of these platforms split the package into parts than you might think - they just don't call the libraries package `libext2fs` like Debian does. It's possible that the other rule needs to be updated to reflect this.

**Comment by amfern** [comment]
I actually needed the libraries only. But adding `e2fsprogs` will work for me, and it will be consistent across other distors

**Review by mabelzhang — APPROVED** [review]

**Review by hidmic — COMMENTED** [review]
LGTM too, but I'll let @cottsay give the final approval.

**Review by clalancette — APPROVED** [review]

**Pull request merged**